### ISSUE TITLE
Trait Detail - Public Vote details

### ISSUE THREAD

**Issue by mcradha** [subhradip82/wesayweb#5]
@subhradip82 

Create service to provide details of public votes given for every trait (both agree and disagree votes)
Service should return the Name of the user who has voted as well as date / time of vote.

**Comment by subhradip82** [comment]
Its already there , need to change the identifier and will push after changing it

**Comment by subhradip82** [comment]
Endpoint :  /userzone/traits/comment/like
Method : Post
Parameter : 
{
 "commentId": 22,
  "like": 1 (0=Dislike, 1= Like)
 }

**Comment by subhradip82** [comment]
Endpoint : /userzone/traits/comment/list
Method : Post 
Parameters : 
{
 "traitIdentifier" : "3eec0d3324520fe5c4a760b1c9ea4d642c3aeb28"
}

Response : 

"response": [
  {
"commentText": "This is a test comment 2",
"commentBy": "Rusty Bose",
"commentDate": "08/Sep/2018 05:51",
"commentId": 22,
"likeCount": 1,
"disLikeCount": 0,
"myLikeDislikeStatus": 1,
"likeDislike": [
  {
"userName": "Rusty Bose",
"likeDislikeDate": "08/Sep/2018 13:34",
"likeOrDislike": 1
}
],
}

**Comment by subhradip82** [comment]
Consider "myLikeDislikeStatus": 2 as i havnt given any vote

**Comment by subhradip82** [comment]
@mcradha  : After verifying please mark the issue as close

**Comment by mcradha** [comment]
Subhra,

I am referring to the names of people who have voted on the Trait, not on
comment.

You have assigned this one to me to verify and close but I am not sure what
to do. Yuvraj, have you integrated the endpoint for this

Regards,
Radha

**Comment by subhradip82** [comment]
I cant see yuvi as the contributor in GIT and thats why i assigned it to you for verification. Can everyone please share his/her git handler so that i can add them as contributor

**Comment by DhakateYuvraj** [comment]
Subhra : my handler on git is 'DhakateYuvraj' (`https://github.com/DhakateYuvraj`).

And this issue is **regarding Trait vote, not comments agree/disagree**
We need a peoples count who gave positive/ negative votes as well as
how many of them voted publicly and anonymously.

But I think it will be **better if you cover this part in traitDetail api**.

So we can consider as a part of issue #10

**Comment by subhradip82** [comment]
Endpoint : /userzone/traits/votes
Parameter : 

{
 "traitIdentifier" : "3eec0d3324520fe5c4a760b1c9ea4d642c3aeb28",
  "traitId" : 484
  
}
---------------- Response--------------

{
"status": "success",
"authtoken": null,
"response": {
"aggreeCount": 0,
"publicAggreeCount": 0,
"annonymousAggreeCount": 0,
"disAggreeCount": 1,
"publicdisAggreeCount": 1,
"annonymousdisAggreeCount": 0,
"aggreeDisAggree": [
  {
"userName": "Rusty Bose",
"likeDislikeDate": "08/Sep/2018 18:11",
"likeOrDislike": 1
}
],
},
"customResponse": null
}

**Comment by subhradip82** [comment]
Endpoint : /userzone/traits/vote/submit

Parameter : 
{
 "traitIdentifier" : "3eec0d3324520fe5c4a760b1c9ea4d642c3aeb28",
  "traitId" : 484,
  "vote" : 1 [1 = Aggree, 0 = DisAggree ],
  "modeOfVote" : 1 [1 = Anonymous, 0 = Public ]
}

**Comment by subhradip82** [comment]
I am just posting the endpoint base , please prefix the server hostname or refer Swagger

**Comment by DhakateYuvraj** [comment]
Previously we was using addTrait api to add vote in user profile page.
End point - "/traitapi/addTrait/"

So do we need to change that to new one ?
Is it capable to identify who is giving vote to whom ? 
In case of voting to self ?

**Comment by subhradip82** [comment]
Yes , traitId denotes the friend for which the trait is related and from loggedin session  we can identify who is giving 

End point - "/traitapi/addTrait/" : Is for adding new trait and the endpoint /userzone/traits/vote/submit is for posting vote for that trait
